### PR TITLE
fixed breathless navigation

### LINKED ISSUE #210: breathless selection not in navigation.
When on the breathless page, selecting one of the options does not move you forward in the navigation. Skipping works fine and takes you to the next page in the navigation.

### PR DESCRIPTION
**Pull request by astewart43-gtri**
closes #210

### CODE REVIEW

File: CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift [hunk 1]
```diff
@@ -69,20 +69,21 @@ class BreathlessViewController: UIViewController {
         
         tableView.rx.itemSelected
             .subscribe(onNext: {indexPath in
+                
                 if indexPath[1] == 0{
-                    print ("0")
+                    self.viewModel.onCauseSelected(cause: .leavingHouseOrDressing)
```
Comment (mellowgeek) [inline CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: Do you need to use self here?

Comment (astewart43-gtri) [reply CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: yes. without it you get "Reference to property 'viewModel' in closure requires explicit 'self.' to make capture semantics explicit"

File: CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift [hunk 2]
```diff
@@ -69,20 +69,21 @@ class BreathlessViewController: UIViewController {
...
-                    print ("2")
+                    self.viewModel.onCauseSelected(cause: .groundOwnPace)
                 }
                 else if indexPath[1] == 3{
-                    print ("3")
+                    self.viewModel.onCauseSelected(cause: .hurryOrHill)
                 }
                 else if indexPath[1] == 4{
-                    print ("4")
+                    self.viewModel.onCauseSelected(cause: .exercise)
```
Comment (ivnsch) [inline CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: The array with the items should be in the view model. In another variable, in the view model, expose it to the view controller (using `Driver`, like in other view models). Here pass only the index: `viewModel.onCauseSelected(indexPath[1])`. In the view model, access the array using the index to get cause.

This is an acceptable solution. I personally prefer to not pass indices around but that would be more involved.

Also, you can declare `viewModel` in the capture list of the closure, to not have to use `self`: `.subscribe(onNext: { [viewModel] indexPath in`

**Review by ivnsch — CHANGES_REQUESTED** [review]

File: CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift (hunk 2 above)
Comment (ivnsch) [reply CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: General takeaways:
- Don't put data and logic in the UI (which includes view controllers). Thus view "model" - it's the backing model for the UI.
- If you're repeating blocks of code for each index/cell, like here, you probably want to access an array instead.
- Orient with existing code: The communication pattern between view models and view controllers is/should be the same everywhere.

Comment (astewart43-gtri) [reply CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: sounds good! will do!

File: CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift [hunk 3]
```diff
@@ -43,15 +43,12 @@ class BreathlessViewController: UIViewController {
...
-        
         tableView.register(cellClass: UITableViewCell.self)
         tableView.tableFooterView = UIView()
         tableView.rowHeight = 70.0
         
         tableView.register(UITableViewCell.self, forCellReuseIdentifier: "Cell")
         
-        let images = ["house", "stop", "ground", "hill", "exercise"]
-        
         Observable.just([L10n.Ux.Breathless.p0, L10n.Ux.Breathless.p1, L10n.Ux.Breathless.p2, L10n.Ux.Breathless.p3, L10n.Ux.Breathless.p4])
```
Comment (ivnsch) [inline CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: This is also data and it should be in the view model.
Also, the translations, images and causes belong together. Orient with e.g. the alerts view model.

Comment (astewart43-gtri) [reply CoEpi/ui/symptomReport/FollowUp/Breathless/BreathlessViewController.swift]: yep, I moved that in the last commit.

**Comment by ivnsch** [comment]
@astewart43-gtri Can this be reviewed (if yes, please don't forget to request review again in the right pane)?

**Comment by astewart43-gtri** [comment]
No. I haven't had time to fix this yet.

**Review by ivnsch — APPROVED** [review]

**Pull request merged**